### PR TITLE
purchase_fiscal_position_update: port to v8 and new API

### PR DESCRIPTION
**Pull request by alexis-via**

### CODE REVIEW

**Comment by rvalyi** [comment]
:+1: LGTM code review, no test

**Comment by pedrobaeza** [comment]
:+1:

File: purchase_fiscal_position_update/purchase.py [hunk 1]
```diff
@@ -51,16 +47,14 @@ def fiscal_position_change(
...
+                product = self.env['product.product'].browse(
+                    line.get('product_id'))
                 taxes = product.supplier_taxes_id
-                tax_ids = fp_obj.map_tax(
-                    cr, uid, fp, taxes, context=context)
-
-                line['taxes_id'] = [(6, 0, tax_ids)]
+                if fp:
+                    taxes = fp.map_tax(taxes)
+                line['taxes_id'] = [(6, 0, [tax.id for tax in taxes])]
```
Comment (yvaucher) [inline purchase_fiscal_position_update/purchase.py]: taxes.ids ? instead of [tax.id for tax in taxes]

**Comment by yvaucher** [comment]
Apart from my inline comment :+1:

File: purchase_fiscal_position_update/purchase.py [hunk 2]
```diff
@@ -22,26 +22,22 @@
...
 
 
-class purchase_order(orm.Model):
+class purchase_order(models.Model):
     _inherit = "purchase.order"
 
-    def fiscal_position_change(
-            self, cr, uid, ids, fiscal_position, order_line,
-            context=None):
+    @api.multi
```
Comment (eLBati) [inline purchase_fiscal_position_update/purchase.py]: Why not `@api.onchange`?

**Comment by alexis-via** [comment]
@eLBati 

I don't know... I took example on odoo/addons/account/account_invoice.py line 1291 : product_id_change() has @api.multi

**Comment by eLBati** [comment]
@alexis-via here is what I propose https://github.com/akretion/purchase-workflow/pull/1

**Comment by lepistone** [comment]
For me it is good but not compulsory to always use new API idioms when porting modules, at least until it stabilizes a bit.
But @eLBati did the job so if you @alexis-via merge @elbati's branch into yours (or you say you have a reason not to do so), I'm happy to merge this PR.

**Comment by alexis-via** [comment]
yes, I'll have a look+test and merge during the OCA code sprint at PyconFR.

**Comment by alexis-via** [comment]
I have accepted the PR of @eLBati with a small modification (no trigger of the on_change with an order line is modified).

But, when working on this, I discovered a very bad bug in the code of the map_tax for the NEW API (the bug is not present in the code for the old API): https://github.com/odoo/odoo/issues/2261

I have made a PR to fix this bug on odoo v8. But, until this bug is fixed on the official branch or in OCB, it is NOT a good idea to use the map_tax function with the new API. So we should delay this merge until the bug is fixed.

**Comment by pedrobaeza** [comment]
Yeah, we discovered it on spanish localization, because we have one fiscal position that maps two destination taxes for one source tax. Let's push to get this fixed.

**Comment by alexis-via** [comment]
For the fix in map_tax with new API: 
Here is the PR on OCB: https://github.com/OCA/OCB/pull/63
Here is the PR on odoo official: https://github.com/odoo/odoo/pull/3316

**Comment by eLBati** [comment]
:+1:

**Pull request merged**